pkg/k8scontext/k8scontext_test.go

```diff
@@ -0,0 +1,190 @@
+package k8scontext_test
+
+import (
+	go_flag "flag"
+	"time"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/getlantern/deepcopy"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	testclient "k8s.io/client-go/kubernetes/fake"
+)
+
+var _ = Describe("K8scontext", func() {
+	var k8sClient kubernetes.Interface
+	var ctxt *k8scontext.Context
+	ingressNS := "test-ingress-controller"
+	ingressName := "hello-world"
+
+	// Create the "test-ingress-controller" namespace.
+	// We will create all our resources under this namespace.
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ingressNS,
+		},
+	}
+
+	// Create the Ingress resource.
+	ingress := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingressName,
+			Namespace: ingressNS,
+			Annotations: map[string]string{
+				k8scontext.IngressClass: k8scontext.IngressControllerName,
+			},
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				v1beta1.IngressRule{
+					Host: "hello.com",
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								v1beta1.HTTPIngressPath{
+									Path: "/hi",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: "hello-world",
+										ServicePort: intstr.IntOrString{
+											Type:   intstr.Int,
+											IntVal: 80,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	go_flag.Lookup("logtostderr").Value.Set("true")
+	go_flag.Set("v", "3")
+
+	BeforeEach(func() {
+		// Create the mock K8s client.
+		k8sClient = testclient.NewSimpleClientset()
+
+		_, err := k8sClient.CoreV1().Namespaces().Create(ns)
+
+		Expect(err).Should(BeNil(), "Unable to create the mock Kuberentes client: %v", err)
+
+		_, err = k8sClient.Extensions().Ingresses(ingressNS).Create(ingress)
+		Expect(err).Should(BeNil(), "Unabled to create ingress resource due to: %v", err)
```

Comment:
nit: typo kuberentes and err msg can be more specific here IMO - "unable to create namespace"